src/index.js

```diff
@@ -1 +1,34 @@
-export default () => true
+import defaultsDeep from 'lodash.defaultsdeep';
+import pickBy from 'lodash.pickby';
+import makeOkCheck from './make-ok-check';
+import buildOptions from './build-options';
+import resolveUrl from './resolve-url';
+
+class Stipulate {
+
+  constructor(options) {
+    this.baseOptions = options;
+  }
+
+  beforeRequest(url, options) {
+    return [url, options];
+  }
+
+  afterResponse(response) {
+    return response;
+  }
+
+  send(urlString, options, query) {
+    const config = buildOptions(options, this.baseOptions);
+    const url = resolveUrl(urlString, query);
+
+    const requestArguments = this.beforeRequest(url, config);
+    const checkOk = makeOkCheck(config);
+
+    return fetch(...requestArguments)
+      .then(checkOk)
+      .then(this.afterResponse);
```

Comment:
>The hooks are isolated from send, and you don't need a default hook that does a passthrough.

your proposed `_before*` and `_after*` functions are passthroughs if `before*` and `after*` don't exist, are they not? I think you're going to have to show me a code example of what you're talking about.

>You wouldn't need to write tests for defaults that don't do anything. Instead, you'd write tests that assert the hooks were called if they were set.

it really sounds like you're proposing a middle man between `send` and the hooks that does the exact same thing that's currently happening.